src/signals/incident/components/form/MapSelectors/Clock/ClockLayer/ClockLayer.tsx

```diff
@@ -19,19 +17,24 @@ export const ClockLayer = () => {
 
   const reportedFeatureType = meta.featureTypes.find(
     ({ typeValue }) => typeValue === 'reported'
-  )
+  ) as ReportedFeatureType
 
-  const reportedFeatures = data.features.filter(
-    (feature) => feature.properties?.meldingstatus === REPORTED
+  const reportedFeatures = data.features.filter((feature) =>
+    Boolean(
+      reportedFeatureType.isReportedField &&
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        feature.properties![reportedFeatureType.isReportedField] ===
+          reportedFeatureType.isReportedValue
+    )
```

Comment:
Filtering of features happens in more places. Can we extract the logic and put it in one place?

Reply:
done